src/libAtomVM/ets_hashtable.c

```diff
@@ -53,14 +53,26 @@ struct EtsHashTable *ets_hashtable_new()
     return htable;
 }
 
+static void ets_hashtable_free_node(struct HNode *node, GlobalContext *global)
+{
+    memory_destroy_heap(node->heap, global);
+    free(node);
+}
+
+void ets_hashtable_free_node_array(struct HNode **allocated, size_t size, GlobalContext *global)
```

Comment:
Nit: I wonder if we should just expose `ets_hashtable_free_node` instead of creating function that just has a loop inside.